File: examples/python/extract.py
"""Extract structured JSON from a page against a schema.

    pip install ghostcrawl
    export GHOSTCRAWL_API_KEY=gck_live_...
    python extract.py

The Python SDK is async — call it from inside an async function.
"""

import asyncio
import os

from ghostcrawl import GhostcrawlClient


async def main() -> None:
    async with GhostcrawlClient(api_key=os.environ["GHOSTCRAWL_API_KEY"]) as client:
        result = await client.extract(
            url="https://example.com",
            schema={
                "type": "object",
                "properties": {
                    "title": {"type": "string"},
                    "summary": {"type": "string"},
                },
            },
        )
        print(result)


if __name__ == "__main__":
    asyncio.run(main())

File: examples/node/extract.mjs
// Extract structured JSON from a page against a schema.
//
//   npm install @ghostcrawl/sdk
//   export GHOSTCRAWL_API_KEY=gck_live_...
//   node extract.mjs

import { createGhostcrawlClient } from '@ghostcrawl/sdk';

const client = createGhostcrawlClient({ token: process.env.GHOSTCRAWL_API_KEY });

const result = await client.extract({
  url: 'https://example.com',
  schema: {
    type: 'object',
    properties: {
      title: { type: 'string' },
      summary: { type: 'string' },
    },
  },
});
// The API returns { url, data, token_estimate }; `data` holds the structured fields.
console.log(result.data ?? result);
